src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -0,0 +1,318 @@
+      const timelock = receive.timelock || null;
+      const address = receive.address || null;
+      proposal.addReceive(
+        req.wallet,
+        token,
+        receive.value,
+        { timelock, address }
+      );
+    }
+
+    res.send({
+      success: true,
+      data: proposal.partialTx.serialize(),
+      isComplete: proposal.partialTx.isComplete(),
+    });
+  } catch (err) {
+    res.send({
+      success: false,
+      error: err.message,
+    });
+  }
+}
+
+async function getMySignatures(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+  const partialTx = req.body.partial_tx;
+
+  // TODO remove this when we create a method to sign inputs in the wallet facade
+  storage.setStore(req.wallet.store);
+  try {
+    const proposal = PartialTxProposal.fromPartialTx(partialTx, network);
+    await proposal.signData('123');
+    res.send({
+      success: true,
+      signatures: proposal.signatures.serialize(),
+    });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  }
+}
+
+async function signTx(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+
+  const partialTx = req.body.partial_tx;
+  const signatures = req.body.signatures || [];
+
+  // TODO remove this when we create a method to sign inputs in the wallet facade
+  storage.setStore(req.wallet.store);
+  try {
+    const tx = assembleTransaction(partialTx, signatures, network);
+
+    res.send({ success: true, txHex: tx.toHex() });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  }
+}
+
+async function signAndPush(req, res) {
+  const validationResult = parametersValidation(req);
+  if (!validationResult.success) {
+    res.status(400).json(validationResult);
+    return;
+  }
+
+  const canStart = lock.lock(lockTypes.SEND_TX);
+  if (!canStart) {
+    res.send({ success: false, error: cantSendTxErrorMessage });
+    return;
+  }
+
+  const network = req.wallet.getNetworkObject();
+
+  const partialTx = req.body.partial_tx;
+  const sigs = req.body.signatures || [];
+
+  // TODO remove this when we create a method to sign inputs in the wallet facade
+  storage.setStore(req.wallet.store);
+  try {
+    const transaction = assembleTransaction(partialTx, sigs, network);
+
+    const sendTransaction = new SendTransaction({ transaction, network });
+    const response = await sendTransaction.runFromMining();
+    res.send({ success: true, ...mapTxReturn(response) });
+  } catch (err) {
+    res.send({ success: false, error: err.message });
+  } finally {
+    lock.unlock(lockTypes.SEND_TX);
+  }
+}
+
+async function getInputData(req, res) {
```

Comment:
When does this method should be used?

Reply:
This is to extract the input data in the atomic-swap format from a transaction hex.
It should be used to extract these signatures made in other ways (e.g. offline signing) but the main reason this API was implemented was to use with MultiSig wallets.

MultiSig wallets have their own signing APIs and finish the process with a partially signed transaction hex (partially since only the P2SH inputs are signed).
This API is meant to extract these signatures so that MultiSIg wallets can send their signatures to an atomic-swap.

Reply:
I think this should be in the docstring of the API

Reply:
Added docstrings